### DISCUSSION TITLE
Possible to retrieve rules like "required"?

### DISCUSSION THREAD

**Discussion by pgsandstrom** [react-hook-form/react-hook-form#4570] [Q&A] [answered]
I am currently integration react-hook-form into a ui library to improve the form handling. I would like to write code like this:

```
<MyTextField
ref={methods.register({ required: true})}
name="organization"
/>
```

Then I would like the component `MyTextField` to detect if it is required or not, so that it can display some visual cues to the user. Is this possible to detect in some way? I want to display these cues before any validation fails, so I cant simply check for errors.

**Comment by bluebill1049** [comment, marked as answer]
have a read here: https://react-hook-form.com/get-started#Integratinganexistingform
